### DISCUSSION TITLE
router events in Next 13

### DISCUSSION THREAD

**Discussion by naeem-qv** [vercel/next.js#47196] [Help]
### Summary

I just want to know what is the replacement for router.events ?? I want to track when the page change started and when it ended, In order to add a loading screen.

### Example

router.events.on()

**Comment by andrepichardo** [comment]
Hi!

According to @timneutkens https://github.com/vercel/next.js/discussions/41745#discussioncomment-3987025, because of concurrent rendering, `router.events` doesn't make sense anymore as any navigation can be canceled without a way to detect that happening. 

The way to listen for page changes would be something like this:

```
'use client'
import {usePathname, useSearchParams} from 'next/navigation'

function useNavigationEvent() {
   const pathname = usePathname()
  const searchParams = useSearchParams()
   useEffect(() => {
      const url = pathname + searchParams.toString()
      sendSomewhere(url)
   }, [pathname, searchParams])
}
```

Hope this helps!

**Reply by khuezy** [reply]
Thanks - how would you catch the event BEFORE? The scenario is I want to use `document.startViewTransition` before next transitions.
